plugins/transport/socket/main.go

```diff
@@ -31,7 +32,9 @@ func rate() int64 {
 }
 
 type configT struct {
-	Path         string `validate:"required"`
+	Path         string
```

Comment:
I believe that one of path or url would be always required. You can achieve that with validate:"required_without=Url" and vice versa

Reply:
Thanks, I didn't know about this.